airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -228,17 +225,18 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         object and returning an updated state object.
         """
         state_value = latest_cursor_value = latest_record.get(self.cursor_field)
+        current_repository = latest_record["repository"]
 
-        if current_stream_state.get(self.repository, {}).get(self.cursor_field):
-            state_value = max(latest_cursor_value, current_stream_state[self.repository][self.cursor_field])
-
-        return {self.repository: {self.cursor_field: state_value}}
+        if current_stream_state.get(current_repository, {}).get(self.cursor_field):
+            state_value = max(latest_cursor_value, current_stream_state[current_repository][self.cursor_field])
+        current_stream_state[current_repository] = {self.cursor_field: state_value}
+        return current_stream_state
```

Comment:
In the variant you proposed, we will lose the `state` for each new repository, since we will return a `new object of the state` of the current repository, overwriting the previous ones.

Reply:
Ok.